src/embedded/embeddedManager.ts

```diff
@@ -141,4 +154,62 @@ export class EmbeddedManager {
   public getUpdateHandlers(): Array<EmbeddedMessageUpdateHandler> {
     return this.updateListeners;
   }
+
+  handleEmbeddedClick(
+    message: any,
+    buttonIdentifier: string | null,
+    clickedUrl: string | null
+  ) {
+    if (clickedUrl && clickedUrl.trim() !== '') {
+      let actionType: string;
+      let actionName: string;
+
+      if (clickedUrl.startsWith(URL_SCHEME_ACTION)) {
+        actionType = URL_SCHEME_ACTION;
+        actionName = clickedUrl.replace(URL_SCHEME_ACTION, '');
+      } else if (clickedUrl.startsWith(URL_SCHEME_ITBL)) {
+        actionType = URL_SCHEME_ITBL;
+        actionName = clickedUrl.replace(URL_SCHEME_ITBL, '');
+      } else {
+        actionType = URL_SCHEME_OPEN;
+        actionName = clickedUrl.replace(URL_SCHEME_OPEN, '');
+      }
+
+      const iterableAction: IterableAction = {
+        type: actionType,
+        data: actionName
+      };
+
+      IterableActionRunner.executeAction(
+        null,
+        iterableAction,
+        IterableActionSource.EMBEDDED
+      );
+    }
+  }
+
+  trackEmbeddedClick(
+    message: any,
+    buttonIdentifier: string | '',
+    clickedUrl: string | ''
```

Comment:
we don't need the ` | ''` since that is just a string.

Reply:
Done